PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/GetSyncPollerOnPollerFluxCheck.java

```diff
@@ -0,0 +1,164 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.LocalInspectionTool;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiExpression;
+import com.intellij.psi.PsiMethodCallExpression;
+import com.intellij.psi.PsiType;
+import com.intellij.psi.util.PsiTreeUtil;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Inspection tool to detect the use of getSyncPoller() on a PollerFlux.
+ * The inspection will check if the method call is on a PollerFlux and if the method call is on an Azure SDK client.
+ * If both conditions are met, the inspection will register a problem with the suggestion to use SyncPoller instead.
+ * <p>
+ * This is an example of an anti-pattern that would be detected by the inspection tool.
+ * public void exampleUsage() {
+ * PollerFlux<String> pollerFlux = createPollerFlux();
+ * <p>
+ * // Anti-pattern: Using getSyncPoller() on PollerFlux
+ * SyncPoller<String, Void> syncPoller = pollerFlux.getSyncPoller();
+ * }
+ */
+public class GetSyncPollerOnPollerFluxCheck extends LocalInspectionTool {
+
+    /**
+     * Method to build the visitor for the inspection tool.
+     *
+     * @param holder Holder for the problems found by the inspection
+     * @return JavaElementVisitor a visitor to visit the method call expressions
+     */
+    @NotNull
+    @Override
+    public JavaElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new GetSyncPollerOnPollerFluxVisitor(holder, isOnTheFly);
+    }
+
+    /**
+     * Visitor class to visit the method call expressions and check for the use of getSyncPoller() on a PollerFlux.
+     * The visitor will check if the method call is on a PollerFlux and if the method call is on an Azure SDK client.
+     */
+    static class GetSyncPollerOnPollerFluxVisitor extends JavaElementVisitor {
+
+        // Instance variables
+        private final ProblemsHolder holder;
+
+        // Define constants for string literals
+        private static final RuleConfig ruleConfig;
+        private static final String ANTI_PATTERN_MESSAGE;
+        private static final String METHOD_TO_CHECK;
+        private static boolean SKIP_WHOLE_RULE;
```

Comment:
Didnt see this in the design, what does this do?

Reply:
it skips the check for the whole rule if the config data couldn't be loaded up for the whole rule